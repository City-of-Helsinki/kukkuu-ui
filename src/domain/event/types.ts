export type FilterValues = {
  date?: string;
  time?: string;
};

type Option = {
  key: string;
  label: string;
  value: string;
};

export type FilterOptions = {
  dates: Option[];
  times: Option[];
};
