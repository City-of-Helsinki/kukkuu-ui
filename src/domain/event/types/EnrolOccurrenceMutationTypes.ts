import { EnrolOccurrenceMutation } from '../../api/generatedTypes/graphql';

export type EnrolOccurrences = NonNullable<
  NonNullable<
    NonNullable<EnrolOccurrenceMutation['enrolOccurrence']>['enrolment']
  >['child']
>['occurrences'];
