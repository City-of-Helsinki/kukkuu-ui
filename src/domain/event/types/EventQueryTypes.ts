import {
  EventOccurrencesFieldsFragment,
  EventOccurrenceFieldsFragment,
} from '../../api/generatedTypes/graphql';

export type Occurrences = EventOccurrencesFieldsFragment;

export type OccurrenceNode = EventOccurrenceFieldsFragment;
