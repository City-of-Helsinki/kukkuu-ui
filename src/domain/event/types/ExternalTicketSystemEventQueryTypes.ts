import {
  TicketmasterEventFieldsFragment,
  LippupisteEventFieldsFragment,
  TixlyEventFieldsFragment,
} from '../../api/generatedTypes/graphql';

export type EventTicketSystem =
  | TicketmasterEventFieldsFragment
  | LippupisteEventFieldsFragment
  | TixlyEventFieldsFragment;
