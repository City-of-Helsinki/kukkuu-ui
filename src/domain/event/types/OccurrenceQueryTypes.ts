import { OccurrenceQuery } from '../../api/generatedTypes/graphql';

export type OccurrenceVenue = NonNullable<
  NonNullable<OccurrenceQuery['occurrence']>['venue']
>;

export type OccurrenceEvent = NonNullable<
  OccurrenceQuery['occurrence']
>['event'];
