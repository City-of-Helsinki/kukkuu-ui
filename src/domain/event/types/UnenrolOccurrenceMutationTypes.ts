import { UnenrolOccurrenceMutation } from '../../api/generatedTypes/graphql';

export type UnenrolOccurrences = NonNullable<
  NonNullable<UnenrolOccurrenceMutation['unenrolOccurrence']>['child']
>['occurrences'];
