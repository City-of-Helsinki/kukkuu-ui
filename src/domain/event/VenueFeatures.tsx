import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { OccurrenceVenue } from './types/OccurrenceQueryTypes';
import Collapsible from '../../common/components/collapsible/Collapsible';

interface VenueFeaturesProps {
  venue: OccurrenceVenue;
}

const VenueFeatures: FunctionComponent<VenueFeaturesProps> = ({ venue }) => {
  const { t } = useTranslation();

  return (
    <>
      {venue.arrivalInstructions && (
        <Collapsible
          item={{
            header: t('venue.features.arrival.heading'),
            body: venue.arrivalInstructions,
          }}
        />
      )}
      {venue.accessibilityInfo && (
        <Collapsible
          item={{
            header: t('venue.features.accessibility.heading'),
            body: venue.accessibilityInfo,
          }}
        />
      )}
      {venue.wcAndFacilities && (
        <Collapsible
          item={{
            header: t('venue.features.wcAndFacilities.heading'),
            body: venue.wcAndFacilities,
          }}
        />
      )}
      {venue.additionalInfo && (
        <Collapsible
          item={{
            header: t('venue.features.additionalInformation.heading'),
            body: venue.additionalInfo,
          }}
        />
      )}
    </>
  );
};

export default VenueFeatures;
