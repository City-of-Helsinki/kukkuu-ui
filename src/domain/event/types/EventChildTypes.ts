import { EnrolOccurrences } from './EnrolOccurrenceMutationTypes';
import { UnenrolOccurrences } from './UnenrolOccurrenceMutationTypes';

export interface ChildEvents {
  childId: string;
  eventIds: string[];
}

export interface ChildOccurrences {
  childId: string;
  occurrences: EnrolOccurrences | UnenrolOccurrences;
}
