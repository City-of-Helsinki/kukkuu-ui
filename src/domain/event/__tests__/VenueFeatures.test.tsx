import VenueFeatures from '../VenueFeatures';
import { customRender as render } from '../../../common/test/customRender';
import { OccurrenceVenue } from '../types/OccurrenceQueryTypes';

const venue: OccurrenceVenue = {
  id: 'auppss',
  name: 'Musiikkitalo',
  address: 'Urho Kekkosen katu 12',
  accessibilityInfo: 'a',
  additionalInfo: 'b',
  arrivalInstructions: 'zz jk Ã¸lk',
  wcAndFacilities: 'zup',
  wwwUrl: 'https://www.venue.com',
};

it('renders snapshot correctly', () => {
  const { container } = render(<VenueFeatures venue={venue} />);
  expect(container).toMatchSnapshot();
});
