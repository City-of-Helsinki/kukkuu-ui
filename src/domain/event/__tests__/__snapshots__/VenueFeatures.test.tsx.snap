// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    class=""
  >
    <button
      aria-controls="collapsible_1"
      aria-expanded="false"
      class="_header_b278db"
    >
      <h3>
        Saapuminen
      </h3>
      <div
        class="_imgWrapper_73ff55 _arrow_b278db"
      >
        <img
          alt=""
          src="/src/assets/icons/svg/angleDown.svg"
        />
      </div>
    </button>
    <div
      aria-hidden="true"
      class="_content_b278db"
      id="collapsible_1"
    >
      <p>
        zz jk ølk
      </p>
    </div>
    <hr />
  </div>
  <div
    class=""
  >
    <button
      aria-controls="collapsible_2"
      aria-expanded="false"
      class="_header_b278db"
    >
      <h3>
        Esteettömyys
      </h3>
      <div
        class="_imgWrapper_73ff55 _arrow_b278db"
      >
        <img
          alt=""
          src="/src/assets/icons/svg/angleDown.svg"
        />
      </div>
    </button>
    <div
      aria-hidden="true"
      class="_content_b278db"
      id="collapsible_2"
    >
      <p>
        a
      </p>
    </div>
    <hr />
  </div>
  <div
    class=""
  >
    <button
      aria-controls="collapsible_3"
      aria-expanded="false"
      class="_header_b278db"
    >
      <h3>
        WC-tilat ja lastenhoitohuone
      </h3>
      <div
        class="_imgWrapper_73ff55 _arrow_b278db"
      >
        <img
          alt=""
          src="/src/assets/icons/svg/angleDown.svg"
        />
      </div>
    </button>
    <div
      aria-hidden="true"
      class="_content_b278db"
      id="collapsible_3"
    >
      <p>
        zup
      </p>
    </div>
    <hr />
  </div>
  <div
    class=""
  >
    <button
      aria-controls="collapsible_4"
      aria-expanded="false"
      class="_header_b278db"
    >
      <h3>
        Lisätietoa
      </h3>
      <div
        class="_imgWrapper_73ff55 _arrow_b278db"
      >
        <img
          alt=""
          src="/src/assets/icons/svg/angleDown.svg"
        />
      </div>
    </button>
    <div
      aria-hidden="true"
      class="_content_b278db"
      id="collapsible_4"
    >
      <p>
        b
      </p>
    </div>
    <hr />
  </div>
</div>
`;
